app/helpers/application_helper.rb

```diff
@@ -11,5 +11,16 @@ def page_entries_info(collection)
       tag.b("#{collection.offset + 1} - #{[collection.per_page * collection.current_page, collection.total_entries].min}") +
       ' of ' + tag.b(collection.total_entries) +
       ' in total'
-end
+  end
+
+  def execute_actions(resource, class_name)
+    api_actions = resource.send(class_name)
+    api_actions.each do |api_action|
+      api_action.execute_action unless api_action.serve_file? || api_action.redirect?
+    end
+  end
+
+  def file_id_from_snippet(file_snippet)
+    ComfortableMexicanSofa::Content::Renderer.new(:page).tokenize(file_snippet).last[:tag_params]
```

Comment:
@Pralish  I wonder where you had to dumpster dive to find this trick

Reply:
I had to go through the cms spec. It took me a while but it was worth it. laughing